internal/app/osd/internal/reg/reg.go

```diff
@@ -237,10 +237,20 @@ func (r *Registrator) Reset(ctx context.Context, in *empty.Empty) (reply *proto.
 
 // Reboot implements the proto.OSDServer interface.
 func (r *Registrator) Reboot(ctx context.Context, in *empty.Empty) (reply *proto.RebootReply, err error) {
+	reply = &proto.RebootReply{}
+
 	// nolint: errcheck
-	unix.Reboot(int(unix.LINUX_REBOOT_CMD_RESTART))
+	defer unix.Reboot(int(unix.LINUX_REBOOT_CMD_RESTART))
 
-	reply = &proto.RebootReply{}
+	return
+}
+
+// Shutdown implements the proto.OSDServer interface.
+func (r *Registrator) Shutdown(ctx context.Context, in *empty.Empty) (reply *proto.ShutdownReply, err error) {
+	reply = &proto.ShutdownReply{}
+
+	// nolint: errcheck
+	defer unix.Reboot(unix.LINUX_REBOOT_CMD_POWER_OFF)
```

Comment:
nice. do these trigger the shutdown stuff in #562

Reply:
not yet, but that should be easy to connect once we get gRPC API for init, this function would call `init->shutdown()` API.